keepassxreboot/keepassxc
# Add tag functionality (Issue #508)

Very early state, just testing the waters to gather feedback.

Add tag functionality (addresses issue #508)

Work on this PR:
- [x] Display tags in entry preview
- [x] Allow modification of tags on entry edit
- [x] Introduce search term "tag:"
- [x]  Implement new side panel, similar to the groups one, with the list of available tags. When clicking on one of those tags, the entry view widget should filter all entries that don't contain the selected tag.
- [x] Display tags as "pills"
- [x] Make tag editing an autocomplete dropdown
- [x] Adapt the styles, make everything beautiful (did the best I could)

Work for subsequent PRs:
- Update the menus with tag actions (still thinking which ones and what actions)
- Delete a tag from the database from the side panel
- When selecting a tag from the side panel filter groups and display only the ones containing entries with the tag
- Check feature/modifications suggestions in comments and Issue #508


## Screenshots

<img width="791" alt="Screenshot 2021-11-24 at 19 57 50" src="https://user-images.githubusercontent.com/867299/143298756-2c401859-caa6-4b3c-8efc-2dd72a340b65.png">

<img width="793" alt="Screenshot 2021-11-24 at 19 54 16" src="https://user-images.githubusercontent.com/867299/143298762-714c72b1-498d-4595-82c8-e336775702f8.png">

![TagsBox2](https://user-images.githubusercontent.com/867299/143298787-90cd0c94-10ca-4161-84d4-8fa5fb818b26.gif)



## Testing strategy
Manually for the fast prototype, will come up with tests next.

## Type of change

- ✅ New feature (change that adds functionality)
- ✅ Documentation (non-code change)
